client/hwlib/src/collectors/os_info.rs

```diff
@@ -91,11 +91,26 @@ impl KernelPackage {
     }
 }
 
-pub(crate) fn get_architecture(runner: &impl CommandRunner) -> Result<String> {
-    let arch = runner
-        .run_command(DPKG, &["--print-architecture"])
-        .with_context(|| "cannot determine system architecture")?;
-    Ok(arch.trim().to_owned())
+pub(crate) fn to_debian_architecture(arch: &str) -> Result<&str> {
+    match arch.trim() {
+        "aarch64" => Ok("arm64"),
+        "arm" => Ok("armhf"),
+        "loongarch64" => Ok("loong64"),
+        "m68k" => Ok("m68k"),
+        "mips" => Ok("mips"),
+        "mips32r6" => Ok("mipsr6"),
+        "mips64" => Ok("mips64"),
+        "mips64r6" => Ok("mips64r6"),
+        "powerpc" => Ok("powerpc"),
+        "powerpc64" => Ok("ppc64el"),
+        "riscv64" => Ok("riscv64"),
+        "s390x" => Ok("s390x"),
+        "sparc" => Ok("sparc"),
+        "sparc64" => Ok("sparc64"),
+        "x86" => Ok("i386"),
+        "x86_64" => Ok("amd64"),
+        _ => Err(anyhow!("cannot convert to debian architecture: {arch:?}")),
```

Comment:
The exhaustiveness is looking nice, I'm not sure about this error form though, perhaps this error message would read slightly more naturally if the `{arch:?}` were in the middle? As in
```rust
"cannot convert {arch:?} to debian architecture"
```